Lombiq.HelpfulLibraries/Libraries/Mvc/TypedRoute.cs

```diff
@@ -59,7 +66,18 @@ public class TypedRoute
             _routeLazy = new Lazy<string>(() =>
                 action.GetCustomAttribute<RouteAttribute>()?.Template is { } route && !string.IsNullOrWhiteSpace(route)
                     ? GetRoute(route)
-                    : $"{area}/{controller.ControllerName()}/{action.GetCustomAttribute<ActionNameAttribute>()?.Name ?? action.Name}");
+                    : $"{_area}/{controller.ControllerName()}/{action.GetCustomAttribute<ActionNameAttribute>()?.Name ?? action.Name}");
+        }
+
+        public string WithLinkGenerator(LinkGenerator linkGenerator, HttpContext httpContext)

```

Comment:
Why not a `ToString()` overload? Also, you could resolve `LinkGenerator` from `HttpContext`.

Reply:
Great idea.